### PR TITLE
Crud regime

### PR DESCRIPTION
**Pull request by jietung**

### CODE REVIEW

File: src/main/java/seedu/exercise/model/regime/UniqueRegimeList.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            if (r.getRegimeName().equals(toCheck.getRegimeName())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Adds an regime to the list.
+     * The person must not already exist in the list.
```
Comment (garylyp) [inline src/main/java/seedu/exercise/model/regime/UniqueRegimeList.java]: Change "person" to "regime"

File: src/main/java/seedu/exercise/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -1,53 +1,25 @@
...
-            + "Example: " + COMMAND_WORD + " 1";
-
-    public static final String MESSAGE_DELETE_EXERCISE_SUCCESS = "Deleted Exercise: %1$s";
+    public static final String MESSAGE_USAGE_EXERCISE = "Parameters: INDEX (must be a positive integer)\n"
+            + "\t\tExample: " + COMMAND_WORD + " " + PREFIX_CATEGORY + "exercise " + PREFIX_INDEX + "1";
 
-    private final Index targetIndex;
+    public static final String MESSAGE_USAGE_REGIME = "Parameters: REGIME NAME, INDEX (must be a positive integer)\n"
+            + "\t\tExample: " + COMMAND_WORD + " " + PREFIX_CATEGORY + "regime "
+            + PREFIX_NAME + "level 1 " + PREFIX_INDEX + "1";
```
Comment (t-cheepeng) [inline src/main/java/seedu/exercise/logic/commands/DeleteCommand.java]: Same here. The `MESSAGE_USAGE` should belong to the specific classes

File: src/main/java/seedu/exercise/model/regime/Regime.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    /**
+     * Returns true if both regimes have the same name.
+     */
+    public boolean isSameRegime(Regime regime) {
+        if (name.equals(regime.getRegimeName())) {
+            return true;
+        }
+        return false;
+    }
```
Comment (t-cheepeng) [inline src/main/java/seedu/exercise/model/regime/Regime.java]: code here can just be devolved into `name.equals(regime.getRegimeName())`. That expression already returns `true/false`

File: src/test/java/seedu/exercise/storage/StorageManagerTest.java [hunk 4]
```diff
@@ -25,8 +25,9 @@
     @BeforeEach
     public void setUp() {
         JsonExerciseBookStorage exerciseBookStorage = new JsonExerciseBookStorage(getTempFilePath("ab"));
+        JsonRegimeBookStorage regimeBookStorage = new JsonRegimeBookStorage(getTempFilePath("cd"));
         JsonUserPrefsStorage userPrefsStorage = new JsonUserPrefsStorage(getTempFilePath("prefs"));
```
Comment (garylyp) [inline src/test/java/seedu/exercise/storage/StorageManagerTest.java]: "ef"

Nothing important here. But I think the first TempFilePath for exerciseBookStorage "ab" might be a short form for 'address-book'.

File: src/main/resources/view/MainWindow.fxml [hunk 5]
```diff
@@ -51,7 +51,7 @@
                 </StackPane>
                 <BorderPane prefHeight="58.0" prefWidth="410.0" styleClass="pane-with-border" VBox.vgrow="ALWAYS">
                     <left>
-                        <StackPane fx:id="sortedListPanelPlaceholder" minHeight="150.0" minWidth="100.0"
+                        <StackPane fx:id="regimeListPanelPlaceholder" minHeight="150.0" minWidth="100.0"
                                    prefHeight="150.0" prefWidth="400.0" BorderPane.alignment="CENTER"/>
```
[the remark this answers is not in the document]
Comment (t-cheepeng) [reply src/main/resources/view/MainWindow.fxml]: its regimes on the left and schedule on the right?

File: src/main/java/seedu/exercise/MainApp.java [hunk 6]
```diff
@@ -75,22 +80,38 @@ public void init() throws Exception {
...
+            initialRegimeData = regimeBookOptional.orElseGet(SampleDataUtil::getSampleRegimeBook);
+        } catch (DataConversionException e) {
+            logger.warning("Data file not in the correct format. Will be starting with an empty RegimeBook");
+            initialRegimeData = new RegimeBook();
+        } catch (IOException e) {
+            logger.warning("Problem while reading from the file. Will be starting with an empty RegimeBook");
+            initialRegimeData = new RegimeBook();
+        }
+
+        return new ModelManager(initialExerciseData, initialRegimeData, userPrefs);
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/java/seedu/exercise/MainApp.java]: ok

File: src/main/java/seedu/exercise/model/regime/UniqueRegimeList.java (hunk 1 above)
Comment (jietung) [reply src/main/java/seedu/exercise/model/regime/UniqueRegimeList.java]: done

File: src/main/java/seedu/exercise/logic/commands/AddCommand.java [hunk 7]
```diff
@@ -1,69 +1,49 @@
...
+
+    public static final String MESSAGE_USAGE_REGIME = "Parameters: "
+            + PREFIX_CATEGORY + "CATEGORY "
+            + PREFIX_NAME + "REGIME NAME "
+            + PREFIX_INDEX + "INDEX\n"
+            + "\t\tExample: " + COMMAND_WORD + " "
+            + PREFIX_CATEGORY + "regime "
+            + PREFIX_NAME + "power set"
+            + PREFIX_INDEX + "1 "
+            + PREFIX_INDEX + "2";
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/java/seedu/exercise/logic/commands/AddCommand.java]: ok

File: src/main/java/seedu/exercise/logic/commands/DeleteCommand.java (hunk 2 above)
Comment (jietung) [reply src/main/java/seedu/exercise/logic/commands/DeleteCommand.java]: ok

File: src/main/java/seedu/exercise/model/regime/Regime.java (hunk 3 above)
Comment (jietung) [reply src/main/java/seedu/exercise/model/regime/Regime.java]: ok

File: src/main/java/seedu/exercise/model/regime/UniqueRegimeList.java [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+     */
+    public void add(Regime toAdd) {
+        requireNonNull(toAdd);
+        internalList.add(toAdd);
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/java/seedu/exercise/model/regime/UniqueRegimeList.java]: ok

File: src/main/java/seedu/exercise/model/ModelManager.java [hunk 9]
```diff
@@ -21,28 +21,30 @@
     private static final Logger logger = LogsCenter.getLogger(ModelManager.class);
 
     private final ExerciseBook exerciseBook;
+    private final RegimeBook regimeBook;
     private final UserPrefs userPrefs;
     private final FilteredList<Exercise> filteredExercises;
-    private final SortedList<Exercise> sortedExercises;
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/java/seedu/exercise/model/ModelManager.java]: Yes. SortedExercises was just filler for the left panel.

File: src/main/java/seedu/exercise/logic/Logic.java [hunk 10]
```diff
@@ -26,20 +28,31 @@
...
     ObservableList<Exercise> getFilteredExerciseList();
+    /**
+     * Returns the RegimeBook.
+     *
+     * @see seedu.exercise.model.Model#getAllRegimeData()
+     */
+    ReadOnlyRegimeBook getRegimeBook();
 
-    ObservableList<Exercise> getSortedExerciseList();
+    ObservableList<Regime> getRegimeList();
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/java/seedu/exercise/logic/Logic.java]: done

File: src/main/resources/view/MainWindow.fxml (hunk 5 above)
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/resources/view/MainWindow.fxml]: The mock up UI in user guide is regime list on the right, left is schedule list.
As for the middle, i think is result display?

File: src/main/resources/view/MainWindow.fxml [hunk 11]
```diff
@@ -51,7 +51,7 @@
                 </StackPane>
                 <BorderPane prefHeight="58.0" prefWidth="410.0" styleClass="pane-with-border" VBox.vgrow="ALWAYS">
                     <left>
-                        <StackPane fx:id="sortedListPanelPlaceholder" minHeight="150.0" minWidth="100.0"
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/resources/view/MainWindow.fxml]: i have changed to `exerciseListPanelPlaceholder` and `regimeListPanelPlaceholder`

File: src/test/java/seedu/exercise/storage/StorageManagerTest.java (hunk 4 above)
Comment (jietung) [reply src/test/java/seedu/exercise/storage/StorageManagerTest.java]: ok

File: src/main/java/seedu/exercise/logic/parser/DeleteCommandParser.java [hunk 12]
```diff
@@ -1,29 +1,75 @@
...
+
+            // index present, delete exercise in regime
+            if (arePrefixesPresent(argMultimap, PREFIX_INDEX)) {
+
+                List<Index> indexes = ParserUtil.parseIndexes(argMultimap.getAllValues(PREFIX_INDEX));
+                return new DeleteRegimeCommand(name, indexes);
+
+            } else { //index not present delete regime
+                return new DeleteRegimeCommand(name, null);
+            }
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/java/seedu/exercise/logic/parser/DeleteCommandParser.java]: done

File: src/main/java/seedu/exercise/logic/parser/AddCommandParser.java [hunk 13]
```diff
@@ -2,56 +2,86 @@
...
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddRegimeCommand.MESSAGE_USAGE));
+            }
+
+            RegimeName regimeName = ParserUtil.parseRegimeName(argMultimap.getValue(PREFIX_NAME).get());
+            List<Index> indexes = ParserUtil.parseIndexes(argMultimap.getAllValues(PREFIX_INDEX));
+
+            return new AddRegimeCommand(indexes, regimeName);
+        }
+
+        throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddExerciseCommand.MESSAGE_USAGE));
```
[the remark this answers is not in the document]
Comment (jietung) [reply src/main/java/seedu/exercise/logic/parser/AddCommandParser.java]: done

File: src/main/resources/view/MainWindow.fxml (hunk 5 above)
[the remark this answers is not in the document]
Comment (t-cheepeng) [reply src/main/resources/view/MainWindow.fxml]: will discuss the finalized UI design on our next meeting then.

[the remark this answers is not in the document]
Comment (garylyp) [reply src/main/resources/view/MainWindow.fxml]: Oh okay I might have remembered wrongly then

**Pull request merged**